main.go

```diff
@@ -412,7 +413,11 @@ func main() {
 	if cfg.Once {
 		err := ctrl.RunOnce(ctx)
 		if err != nil {
-			log.Fatal(err)
+			if strings.Contains(err.Error(), "Throttling: Rate exceeded") {
+				log.Error(err)
+			} else {
+				log.Fatal(err)
+			}
```

Comment:
This is the wrong place to handle it. This part of code is executed when we want ExternalDNS to run a single synchronisation and then exit (`--once`). We should keep the `log.Fatal` here.

The problem you want to solve needs to be addressed in the `Controller`'s `Run()` function: [This is the ](https://github.com/kubernetes-sigs/external-dns/blob/master/controller/controller.go#L314)

Reply:
I'll look into it. Also saw your comments below, they are helpful. Thanks.